File: core/src/main/java/com/sequenceiq/cloudbreak/repository/CloudbreakEventSpecifications.java
package com.sequenceiq.cloudbreak.repository;

import java.util.Date;

import org.springframework.data.jpa.domain.Specification;

import com.sequenceiq.cloudbreak.domain.CloudbreakEvent;

public class CloudbreakEventSpecifications {

    private CloudbreakEventSpecifications() {
    }

    public static Specification<CloudbreakEvent> eventsForUser(final String user) {
        return (cloudbreakEventRoot, query, cb) -> cb.equal(cloudbreakEventRoot.get("owner"), user);
    }

    public static Specification<CloudbreakEvent> eventsSince(final Long since) {
        return (cloudbreakEventRoot, query, cb) -> {
            if (since != null) {
                return cb.greaterThanOrEqualTo(cloudbreakEventRoot.get("eventTimestamp"), new Date(since));
            } else {
                return cb.and();
            }
        };
    }

}

File: core/src/main/java/com/sequenceiq/cloudbreak/repository/CloudbreakEventSpecifications.kt
package com.sequenceiq.cloudbreak.repository

import java.util.Date

import javax.persistence.criteria.CriteriaBuilder
import javax.persistence.criteria.CriteriaQuery
import javax.persistence.criteria.Predicate
import javax.persistence.criteria.Root

import org.springframework.data.jpa.domain.Specification

import com.sequenceiq.cloudbreak.domain.CloudbreakEvent

object CloudbreakEventSpecifications {

    fun eventsForUser(user: String): Specification<CloudbreakEvent> {
        return Specification { cloudbreakEventRoot, query, cb -> cb.equal(cloudbreakEventRoot.get<Any>("owner"), user) }
    }

    fun eventsSince(since: Long?): Specification<CloudbreakEvent> {
        return Specification { cloudbreakEventRoot, query, cb ->
            if (since != null) {
                cb.greaterThanOrEqualTo(cloudbreakEventRoot.get<Date>("eventTimestamp"), Date(since))
            } else {
                cb.and()
            }
        }
    }

}
